src/config_generator/from_openapi.rs

```diff
@@ -0,0 +1,209 @@
+}
+
+fn make_config_types(spec: &Spec) -> BTreeMap<String, Type> {
+    let components = spec.components.iter().next().cloned().unwrap();
+    let mut types = BTreeMap::new();
+    let mut fields = BTreeMap::new();
+
+    for (mut path, path_item) in spec.paths.clone().into_iter() {
+        let (method, operation) = [
+            (Method::GET, path_item.get),
+            (Method::HEAD, path_item.head),
+            (Method::OPTIONS, path_item.options),
+            (Method::TRACE, path_item.trace),
+            (Method::PUT, path_item.put),
+            (Method::POST, path_item.post),
+            (Method::DELETE, path_item.delete),
+            (Method::PATCH, path_item.patch),
+        ]
+        .into_iter()
+        .filter_map(|(method, operation)| operation.map(|operation| (method, operation)))
+        .next()
+        .unwrap();
+        let output_type = operation
+            .responses
+            .get("200")
+            .unwrap()
+            .resolve(spec)
+            .unwrap()
+            .content
+            .get("application/json")
+            .cloned()
+            .unwrap()
+            .schema
+            .unwrap();
+
+        let args: BTreeMap<String, Arg> = operation
+            .parameters
+            .iter()
+            .map(|param| {
+                let param = param.resolve(spec).unwrap();
+                let (is_list, name) =
+                    get_schema_type(|| param.param_type.unwrap(), param.schema.unwrap());
+                (
+                    param.name,
+                    Arg {
+                        type_of: name,
+                        list: is_list,
+                        required: param.required.unwrap_or_default(),
+                        doc: None,
+                        modify: None,
+                        default_value: None,
+                    },
+                )
+            })
+            .collect();
+
+        let schema = output_type.resolve(spec).unwrap();
+
+        let (is_list, name) = get_schema_type(
+            || {
+                if let ObjectOrReference::Ref { ref_path } = output_type {
+                    ref_path.split('/').last().unwrap().to_string()
+                } else {
+                    unreachable!()
+                }
+            },
+            schema,
+        );
+
+        if !args.is_empty() {
+            let re = regex::Regex::new(r"\{\w+\}").unwrap();
+            path = re
+                .replacen(path.as_str(), 0, |cap: &regex::Captures| {
+                    let arg_name = &cap[0][1..cap[0].len() - 1];
+                    format!("{{{{args.{}}}}}", arg_name)
+                })
+                .to_string();
+        }
+
+        let field = Field {
+            type_of: name,
+            list: is_list,
+            args,
+            http: Some(Http { path, method, ..Default::default() }),
+            ..Default::default()
+        };
+
+        fields.insert(operation.operation_id.unwrap(), field);
```

Comment:
Split the function into smaller functions. One of building the type, and one for building the field for a type.